### ISSUE TITLE
Make tlog vdisk goroutine lock free

### ISSUE THREAD

**Issue by iwanbk** [zero-os/0-Disk#241]
I had implemented lock free (and hopefully race free) implementation of tlog vdisk goroutine.
We need the it to protect `vdisk.expectedSequence` variable.
It's been always failed in nbdserver tests.

It should be implemented again after #240 getting merged to master.
The `expectedSequence` variable should be made local to the goroutine using it.

**Comment by GlenDC** [comment]
Why is this moved to the Roadmap? Seems like an important enough issue to fix ASAP.

**Comment by iwanbk** [comment]
> Seems like an important enough issue to fix ASAP.

It is not an issue. I thought we can remove the lock/mutex in vdisk's goroutine easily and only use channel for communication.

I have tried it, it was not as easy as i thought.

**Comment by GlenDC** [comment]
Yeah never mind, priorities have changed, and we have a lot on our plate. So I think having this issue on the roadmap is good enough for now.
